src/inversify.config.ts

```diff
@@ -0,0 +1,114 @@
+import { Container, interfaces } from "inversify";
+import { buildProviderModule } from "inversify-binding-decorators";
+import {
+  Uri,
+  Event,
+  EventEmitter,
+  WorkspaceFolder,
+  CancellationToken,
+} from "vscode";
+import { CommandProcessExecution } from "./dbt_client/commandProcessExecution";
+import { GraphMetaMap } from "./domain";
+import { DBTProject } from "./manifest/dbtProject";
+import { DbtProjectContainer } from "./manifest/dbtProjectContainer";
+import { DBTWorkspaceFolder } from "./manifest/dbtWorkspaceFolder";
+import { ManifestCacheChangedEvent } from "./manifest/event/manifestCacheChangedEvent";
+import { ProjectConfigChangedEvent } from "./manifest/event/projectConfigChangedEvent";
+import { DBTProjectLog } from "./manifest/handlers/dbtProjectLog";
+import { SourceFileWatchers } from "./manifest/handlers/sourceFileWatchers";
+import { TargetWatchers } from "./manifest/handlers/targetWatchers";
+import { ModelTreeviewProvider } from "./treeview_provider/ModelParentTreeviewProvider";
+
+export const container = new Container();
+container.load(buildProviderModule());
+
+container
```

Comment:
if only the eventemitter is required, these might better be moved into a singleton and just provided by the default mechanism